Gems/Atom/Feature/Common/Assets/Materials/Types/MaterialInputs/EyeInput.azsli

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include "MaterialInputs/BaseColorInput.azsli"
+#include "MaterialInputs/NormalInput.azsli"
+#include "MaterialInputs/RoughnessInput.azsli"
+
+// This file provides utilities for common handling of inputs for eye-specific 
+// parameters of Eye materials.
+
+// These macros can be used to declare common shader inputs for this feature. 
+// Use the COMMON_SRG_INPUTS_* macro in your material SRG definition, and use
+// the COMMON_OPTIONS_* macro at the global scope in your shader. Then you can
+// pass these variables to the Get*Input() function below.
+// You can optionally provide a prefix for the set of inputs which corresponds 
+// to a prefix string supplied by the .materialtype file. This is common for 
+// multi-layered material types.
+
+// Since we are using nested property groups (iris and sclera contain separate 
+// base color, normal, and roughness groups), we identify each subgroup with a 
+// prefix (see Eye.materialtype).
+// In this case, each iris'/sclera's subgroup property 'x' will be identified 
+// as m_iris_x/m_sclera_x.
+// e.g. the base color for the iris would be identified in the Material SRG as
+// 'm_iris_m_baseColor'.
+#define COMMON_SRG_INPUTS_EYE(prefix) \
+COMMON_SRG_INPUTS_BASE_COLOR(m_iris_) \
+COMMON_SRG_INPUTS_NORMAL(m_iris_) \
+COMMON_SRG_INPUTS_ROUGHNESS(m_iris_) \
+COMMON_SRG_INPUTS_BASE_COLOR(m_sclera_) \
+COMMON_SRG_INPUTS_NORMAL(m_sclera_) \
+COMMON_SRG_INPUTS_ROUGHNESS(m_sclera_)
+
+#define COMMON_OPTIONS_EYE(prefix) \
+COMMON_OPTIONS_BASE_COLOR(o_iris_) \
+COMMON_OPTIONS_NORMAL(o_iris_) \
+COMMON_OPTIONS_ROUGHNESS(o_iris_) \
+COMMON_OPTIONS_BASE_COLOR(o_sclera_) \
+COMMON_OPTIONS_NORMAL(o_sclera_) \
+COMMON_OPTIONS_ROUGHNESS(o_sclera_)
```

Comment:
As mentioned above, you can just inline these macros and get rid of the EyeInput.azsli file. 

The pattern you are following here is like "MaterialInputs/BaseColorInput.azsli". That pattern is used for reusing SRG inputs and getter functions like GetBaseColorInput, making it easier for those things to be used in multiple material types. I doubt Eye shading will be used across multiple material types, so we don't need that pattern here, it's just making the structure more complicated.

Reply:
Makes sense, thanks for pointing out!